client/src/composables/toolTrainingMaterial.ts

```diff
@@ -0,0 +1,168 @@
+import { useConfig } from "./config";
+import { computed, ref, watch, type Ref } from "vue";
+
+type Name = string;
+type Repo = string;
+type ID = string;
+
+type TrainingId = `${Repo}/${Name}` | ID;
+
+type TrainingDetails = {
+    tool_id: Array<
+        [
+            string, // ID (unused)
+            string // Version
+        ]
+    >;
+    tutorials: Array<
+        [
+            string, // tutorial ID (unused)
+            string, // Title
+            string, // Category
+            string // URL
+        ]
+    >;
+};
+
+type TrainingMaterialResponse = {
+    [id: TrainingId]: TrainingDetails;
+};
+
+type Config = {
+    tool_training_recommendations: boolean;
+    tool_training_recommendations_api_url: string;
+    tool_training_recommendations_link: string;
+};
+
+export type TutorialDetails = {
+    category: string;
+    title: string;
+    url: URL;
+};
+
+const cachedResponse: Ref<TrainingMaterialResponse | null> = ref(null);
+
+export function useToolTrainingMaterial(id: ID, name: Name, version: string, repo?: Repo) {
+    const { config, isLoaded }: { config: Ref<Config>; isLoaded: Ref<boolean> } = useConfig();
+    const apiEnabled = computed(() => {
+        return Boolean(
+            isLoaded.value &&
+                config.value.tool_training_recommendations &&
+                config.value.tool_training_recommendations_api_url
+        );
+    });
+
+    watch(
+        () => isLoaded.value,
+        async () => {
+            if (!isLoaded.value) {
+                return;
+            }
+
+            if (apiEnabled.value && !cachedResponse.value) {
+                const res = await fetch(config.value.tool_training_recommendations_api_url);
+
+                if (res.ok) {
+                    cachedResponse.value = await res.json();
+                }
+            }
+        },
+        { immediate: true }
+    );
+
+    const identifier = computed(() => {
+        if (repo) {
+            return `${repo}/${name.toLowerCase()}`;
```

Comment:
The tool name can be completely independent of the tool id

Reply:
Just saw you mentioned bwa_mem above. I don't see any way of accessing the identifier (`devteam/bwa_mem`) for this with the info currently in the tool form. Do you know where this could be retrieved?

Reply:
If you want to match against https://training.galaxyproject.org/training-material/api/top-tools.json I'd work with the tool id. That should either give you an exact match, or you can chop off the version by splitting on the last `/`